### PR TITLE
Fix a bug where last edge is not locked in WRITE_INODE pattern

### PR DESCRIPTION
**Pull request by cheng-chang**
Suppose WRITE_INODE on path /a/b/c, and only /a exists, then both inode a and edge a->b should be read locked, otherwise, a concurrent rename from /a/c -> a/b will succeed.

See the enhanced unit test in LockedInodePathTest.

### CODE REVIEW

File: core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java [hunk 1]
```diff
@@ -159,6 +159,26 @@
...
+     *
+     * path to lock: /a/b/c
+     * existing inodes: /a/b/c
+     * last edge: b->c
+     *
+     * path to lock: /a/b/c
+     * existing inodes: /a
+     * last edge: a->b
+     *
+     * @return whether the last edge should be locked
```
Comment (gpang) [inline core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java]: write locked?

File: core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java [hunk 2]
```diff
@@ -159,6 +159,26 @@
...
+     */
+    public boolean shouldLockLastEdge() {
```
Comment (gpang) [inline core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java]: should this be named "writeLockLastEdge"?

**Comment by cheng-chang** [comment]
@gpang PTAL

**Comment by cheng-chang** [comment]
alluxio-bot, merge this please

**Pull request merged**